net/ProfileViewController.m

```diff
@@ -426,4 +387,62 @@ - (IBAction)changeUserDatails:(UIButton *)sender {
     }
 }
 
+#pragma mark - Keyboard
+- (void)registerForKeyboardNotifications
+{
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(keyboardDidShown:)
+                                                 name:UIKeyboardDidShowNotification object:nil];
+    
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(keyboardWillBeHidden:)
+                                                 name:UIKeyboardWillHideNotification object:nil];
+    
+}
+
+- (void)keyboardDidShown:(NSNotification*)aNotification
+{
+    NSDictionary* info = [aNotification userInfo];
+    CGSize kbSize = [[info objectForKey:UIKeyboardFrameBeginUserInfoKey] CGRectValue].size;
+    
+    
+    CGRect aRect = self.view.frame;
+    aRect.size.height -= kbSize.height;
+    
+    UIEdgeInsets contentInsets = UIEdgeInsetsMake(80, 0.0, aRect.size.height, 0.0);
+    self.scrollView.contentInset = contentInsets;
+    self.scrollView.scrollIndicatorInsets = contentInsets;
+
+    CGPoint point = [self.activeField.superview convertPoint:self.activeField.frame.origin toView:nil];
+    
+    point.y += self.activeField.frame.size.height;
+    
+    if (!CGRectContainsPoint(aRect, point) ) {
+        [self.scrollView setContentOffset:CGPointMake(0.0, point.y - aRect.size.height) animated:YES];
+    }
+}
+
+- (void)keyboardWillBeHidden:(NSNotification*)aNotification
+{
+    UIEdgeInsets contentInsets = UIEdgeInsetsZero;
+    contentInsets.top = 80;
+    self.scrollView.contentInset = contentInsets;
+    self.scrollView.scrollIndicatorInsets = contentInsets;
+}
+
+- (void)textFieldDidBeginEditing:(UITextField *)textField
+{
+    self.activeField = textField;
+}
+
+- (void)textFieldDidEndEditing:(UITextField *)textField
+{
+    self.activeField = nil;
+}
+
+- (IBAction)dismissKeyboard:(id)sender
+{
+    [self.activeField resignFirstResponder];
+}
```

Comment:
it is have sense to set activeField to nil. May be its not necessarily but imho will be good )